java/bulk-export/bulk-export-core/src/test/java/sleeper/bulkexport/model/BulkExportLeafPartitionQuerySerDeTest.java

```diff
@@ -0,0 +1,356 @@
+import sleeper.core.properties.testutils.FixedTablePropertiesProvider;
+import sleeper.core.range.Range.RangeFactory;
+import sleeper.core.range.Region;
+import sleeper.core.schema.Field;
+import sleeper.core.schema.Schema;
+import sleeper.core.schema.type.IntType;
+import sleeper.core.schema.type.StringType;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static sleeper.core.properties.table.TableProperty.TABLE_ID;
+import static sleeper.core.properties.table.TableProperty.TABLE_NAME;
+import static sleeper.core.properties.testutils.InstancePropertiesTestHelper.createTestInstanceProperties;
+import static sleeper.core.properties.testutils.TablePropertiesTestHelper.createTestTablePropertiesWithNoSchema;
+
+public class BulkExportLeafPartitionQuerySerDeTest {
+    private final Field field = new Field("key", new IntType());
+    private final Schema schema = Schema.builder()
+            .rowKeyFields(field)
+            .valueFields(new Field("value1", new StringType()), new Field("value2", new StringType()))
+            .build();
+    private final InstanceProperties instanceProperties = createTestInstanceProperties();
+    private final TableProperties tableProperties = createTestTablePropertiesWithNoSchema(instanceProperties);
+    private final String tableId = tableProperties.get(TABLE_ID);
+    private final String exportId = UUID.randomUUID().toString();
+    private final String subExportId = UUID.randomUUID().toString();
+    private final String leafPartitionId = UUID.randomUUID().toString();
+
+    @Test
+    public void shouldSerDeBulkExportLeafPartitionQuery() {
+        // Given
+        RangeFactory rangeFactory = new RangeFactory(schema);
+        Region region1 = new Region(rangeFactory.createRange(field, 1, true, 10, true));
+        Region partitionRegion = new Region(rangeFactory.createRange(field, 0, 1000));
+        BulkExportLeafPartitionQuery bulkExportLeafPartitionQuery = BulkExportLeafPartitionQuery.builder()
+                .tableId(tableId)
+                .exportId(exportId)
+                .subExportId(subExportId)
+                .regions(List.of(region1))
+                .leafPartitionId(leafPartitionId)
+                .partitionRegion(partitionRegion)
+                .files(Collections.singletonList("/test/file.parquet"))
+                .build();
+
+        BulkExportLeafPartitionQuerySerDe querySerDe = generateQuerySerDe("my-table", schema, true);
+
+        // When
+        String json = querySerDe.toJson(bulkExportLeafPartitionQuery, true);
+
+        BulkExportLeafPartitionQuery deserialisedQuery = querySerDe.fromJson(json);
+
+        // Then
+        String expectedJson = "{\n" +
+                "  \"tableId\": \"" + tableId + "\",\n" +
+                "  \"exportId\": \"" + exportId + "\",\n" +
```

Comment:
Since the JSON is pretty big now, for the tests in this file it seems worth either using a multiline string literal or putting the JSON in a separate file, e.g. with `Approvals.verify`, as in `CompactionJobIdAssignmentCommitRequestSerDeTest`.

I'd encourage you to try `Approvals.verify`. If you prefer using a multiline string literal here's some information about that:
https://docs.oracle.com/en/java/javase/17/text-blocks/index.html
https://www.baeldung.com/java-text-blocks

Reply:
Updated to use Approvals.verify